### PR TITLE
Quick fix for https://github.com/fluttercommunity/backdrop/issues/83

### PR DESCRIPTION
**Pull request by felixwortmann**
This could be a fix for #83. <br>
It should be further testet though.

### CODE REVIEW

File: lib/scaffold.dart [hunk 1]
```diff
@@ -424,7 +420,7 @@ class BackdropScaffoldState extends State<BackdropScaffold>
 
   @override
   void dispose() {
-    if (_shouldDisposeAnimationController) _animationController.dispose();
+    if(_animationController != null) _animationController.dispose();
```
Comment (WieFel) [inline lib/scaffold.dart:423]: This could be simplified further to `_animationController?.dispose();` ;)

Comment (felixwortmann) [reply lib/scaffold.dart]: I just did

File: lib/scaffold.dart [hunk 2]
```diff
@@ -424,8 +420,8 @@ class BackdropScaffoldState extends State<BackdropScaffold>
 
   @override
   void dispose() {
+    _animationController?.dispose();
```
Comment (daadu) [inline lib/scaffold.dart:423]: The bug is not with `_shouldDisposeAnimationController`, it is calling of `_animationController?.dispose();` before `super.dispose` here, that has fixed the issue.

I suggest reverting back the `_shouldDisposeAnimationController` logic if the above is correct.

File: lib/scaffold.dart [hunk 3]
```diff
@@ -424,8 +420,8 @@ class BackdropScaffoldState extends State<BackdropScaffold>
...
     super.dispose();
```
Comment (daadu) [inline lib/scaffold.dart:424]: The bug is not with `_shouldDisposeAnimationController`, it is calling of `_animationController?.dispose();` before `super.dispose` here, that has fixed the issue.

I suggest reverting back the `_shouldDisposeAnimationController` logic if the above is correct.

Comment (daadu) [reply lib/scaffold.dart:424]: Also, add a single line comment above `_shouldDisposeAnimationController` explaining its logic, something like `// not disposing of _animationController, if it was passed in widget.`

Comment (daadu) [reply lib/scaffold.dart:424]: Also the following:
```dart
if (widget.animationController == null && widget.controller == null) {
      _shouldDisposeAnimationController = true;
    }
```

could be simplified to:

```dart
// not disposing of `_animationController`, if it was passed in widget.
 _shouldDisposeAnimationController = (widget.animationController ?? widget.controller) == null;
```

**Comment by daadu** [comment]
This fix/feature has landed with [v0.5.3 on pub.dev](https://pub.dev/packages/backdrop/versions/0.5.3).

**Pull request merged**